workflow/util/pod_name.go

```diff
@@ -57,7 +57,7 @@ func GeneratePodName(workflowName, nodeName, templateName, nodeID string, versio
 	}
 
 	prefix := workflowName
-	if !strings.Contains(nodeName, ".inline") {
+	if !strings.Contains(nodeName, ".inline") && templateName != "" {
 		prefix = fmt.Sprintf("%s-%s", workflowName, templateName)
```

Comment:
```suggestion
	if templateName != "" {
		prefix = fmt.Sprintf("%s-%s", prefix, templateName)
```

confirmed that this works and will match the UI as well -- see also #12964 .
an inline template will have no `templateName`, so it is equivalent to the previous logic.

also changed the `prefix` assignment a bit to simplify it

Reply:
Resolved in #13261